### ISSUE TITLE
[build_images]  Add support for resolve secrets from a Azure KeyVault / Parameter Store

### ISSUE THREAD

**Issue by alambike** [prefapp/firestarter-workflows#13]
### Motivation

We would like to be able to resolve secrets, in the `build_images` workflow config file, so any secret you need to be used in the building of the final image, can be available to the build process, in the recommended way:

https://docs.docker.com/build/ci/github-actions/secrets/

https://www.youtube.com/watch?v=IzdWk6nA_Ho ❗ ❗ 
[OIDC con AWS](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-azure#updating-your-github-actions-workflow)
[OIDC con Azure](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-azure#updating-your-github-actions-workflow)

### Acceptance criteria

**Comment by itasahobby** [comment]
We are currently using [build](https://dagger-io.readthedocs.io/en/sdk-python-v0.6.0/api.html#dagger.api.gen.Container.build) method:

https://github.com/prefapp/firestarter-workflows/blob/a0bffc919346a1faf804efa824898eaafaecedf5/firestarter/workflows/build_images/build_images.py#L104-L110

It already supports using secrets in one of the parameters
> secrets – Secrets to pass to the build. They will be mounted at /run/secrets/[secret-name]

Also we allow workflows to use secrets in our [run-dagger-py](https://github.com/prefapp/run-dagger-py) composite action.

https://github.com/prefapp/run-dagger-py/blob/a547e7d664ab0e9132fa27344332eb5f478520c7/action.yaml#LL52C6-L52C6

**Comment by itasahobby** [comment]
Created a simple test to check how dagger handles secrets while using a custom Dockerfile, check the code below:
```python
import sys

import anyio

import dagger


async def test():
    async with dagger.Connection(dagger.Config(log_output=sys.stderr)) as client:

        src = client.host().directory(".")

        secret_token = client.set_secret("ghApiToken", "TOKEN")

        image = await client.container().build(context=src, dockerfile="./Dockerfile", secrets=[secret_token])

        image = image.with_exec(["env"])

        out = await image.stdout()

        print(out)

if __name__ == "__main__":
    anyio.run(test)

```

However it does not display the secret either in environment variables or in `/run/secrets`.

**Comment by alambike** [comment]
Anyway this issue was about how to deal with **resolving secrets from an external vault** (Azure Keyvault, AWS Parameter store...)

The secret in GitHub would take the coordinates to a secret in an external vault, and the CI or dagger would have to have the logic and sufficient permissions to be able to access that vault in the provider and decrypt the secret, preferably, through OIDC.

**Comment by alambike** [comment]
@itasahobby , I reopen this issue until add support for Azure Keyvault

**Comment by alambike** [comment]
Closed by #31
